llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Transforms/Transforms.cpp

```diff
@@ -904,19 +903,59 @@ struct LinalgExtPackOpVectorizationPass
 
   void runOnOperation() override {
     MLIRContext *ctx = &getContext();
+    // Apply tiling to make outer dims be all 1s.
+    {
+      RewritePatternSet patterns(ctx);
+      auto tilingOptions =
+          linalg::LinalgTilingOptions().setTileSizeComputationFunction(
+              [](OpBuilder &builder, Operation *op) {
+                Location loc = op->getLoc();
+                auto packOp = cast<PackOp>(op);
+                auto innerDims =
+                    extractFromI64ArrayAttr(packOp.getInnerDimsPos());
+                int inputRank = packOp.getInputRank();
+                SmallVector<Value> tileSizes(
+                    inputRank, builder.create<arith::ConstantIndexOp>(loc, 1));
+                return tileSizes;
+              });
+      auto funcOp = getOperation();
+      patterns.add<TilingInterfaceTilingPattern>(
```

Comment:
Please dont use this pattern. I want to deprecate it. Instead of using the pattern directly call `scf::tileUsingSCFForOp`...

Reply:
I'll give it a shot, thanks for the pointer!

Reply:
After studying the code more, I found that using scf::tileUsingSCFForOp should be a migration in LinalgExt tiling pattern. The method should be used by a pattern, while here is constructing the pass.

It is a helper for patterns. It is not a pattern. The method needs a rewriter. We'll need to implement a pattern to use it. In this PR, we're populating the LinalgExt tiling pattern, which is a layer above using the method.

```
FailureOr<scf::SCFTilingResult>
mlir::scf::tileUsingSCFForOp(RewriterBase &rewriter, TilingInterface op,
                             const scf::SCFTilingOptions &options)
```

Reply:
But we don't need to use pattern rewriter right? We can just do a walk and apply the Tiling once? Basically trying to avoid use of the markers that is deprecated.

Reply:
Got it, I see the point now! I have a prototype doing it, but that needs some changes upstream (for handling size=1 in tiling) like what've been done in the PR. I'll send it out for review.